python/amici/ode_export.py

```diff
@@ -1258,14 +1258,16 @@ def _generate_symbol(self, name: str) -> None:
             ])
             return
         elif name == 'dtcldp':
+            # check, whether the CL consists of only one state. Then,
+            # sensitivities drop out, syms can be deleted
+            state_set = {state.get_id() for state in self._states}
             self._syms[name] = sp.Matrix([
-                [
-                    sp.Symbol(f's{strip_pysb(tcl.get_id())}__'
-                              f'{strip_pysb(par.get_id())}',
-                              real=True)
+                [sp.Symbol(f's{strip_pysb(tcl.get_id())}__'
+                           f'{strip_pysb(par.get_id())}', real=True)
                     for par in self._parameters
-                ]
-                for tcl in self._conservationlaws
+                 ]
+                for tcl in self._conservationlaws if len(
```

Comment:
implement check as function, e.g. `conservation_has_multispecies`? Inherently adds documentation through the function name and simplifies maintenance by reducing duplication.

Reply:
Maybe easier to read than a 6-line list comprehension indeed...